CriticalMass/TextFieldWithInsets.swift

```diff
@@ -7,9 +7,64 @@
 
 import UIKit
 
+/// extension to colorize all subviews except the placeholderView
+private extension UITextField {
+    func setEditorBackgroundColor(to color: UIColor?) {
+        subviews.forEach { view in
+            if view is UILabel {
+                return
+            }
+            view.backgroundColor = color
+        }
+    }
+
+    var placeholderColor: UIColor {
+        get {
+            return attributedPlaceholder?.attribute(.foregroundColor, at: 0, effectiveRange: nil) as? UIColor ?? .clear
+        }
+        set {
+            guard let attributedPlaceholder = attributedPlaceholder else { return }
+            let attributes: [NSAttributedString.Key: UIColor] = [.foregroundColor: newValue]
+            self.attributedPlaceholder = NSAttributedString(string: attributedPlaceholder.string, attributes: attributes)
+        }
+    }
+}
+
 class TextFieldWithInsets: UITextField {
     var insets: UIEdgeInsets = .zero
 
+    @objc
+    dynamic var textFieldBackgroundColor: UIColor? {
+        willSet {
+            backgroundColor = newValue
+        }
+    }
+
+    @objc
+    dynamic var placeholderTextColor: UIColor! {
+        didSet {
+            placeholderColor = placeholderTextColor
+        }
+    }
+
+    // sets the backgroundColor when input begins
+    override func becomeFirstResponder() -> Bool {
+        let didBecomeFirstResponder = super.becomeFirstResponder()
+        if didBecomeFirstResponder {
+            setEditorBackgroundColor(to: textFieldBackgroundColor)
```

Comment:
Why is that needed? I don't see any difference in the backgroundColor after becoming or resigning first responder

Reply:
This is a workaround since otherwise activating the textfield always changed the backgroundColor of it when it became first responder. Took me some time but that was the only solution I found. I could comment it for better understanding why this code is there or maybe you know another option.

Reply:
yes it would be nice to add comments why this hack is needed